Arriba/Arriba.Web/parts/Grid.jsx

```diff
@@ -330,34 +320,21 @@ export default class Grid extends EventedComponent {
     }
     componentDidMount() {
         super.componentDidMount();
-        var defaultQuery = (this.props.params.p === "default" ? configuration.gridDefault : this.props.params.p);
-        if (defaultQuery) {
+        var defaultQuery = this.props.params.p || configuration.gridDefault;
+        if (!this.props.query && defaultQuery) {
             this.selectDefaultQuery(defaultQuery);
         }
         this.componentDidUpdate({}, {});
     }
     componentDidUpdate(prevProps, prevState) {
+        const diffProps = Object.diff(prevProps, this.props);
         const diffState = Object.diff(prevState, this.state);
 
-        if (diffState.hasAny("query")) {
-            // Only query every 250 milliseconds while typing
-            this.timer = this.timer || window.setTimeout(() => this.setState({ debouncedQuery: this.state.query }), 250);
-        }
-
-        if (diffState.hasAny("debouncedQuery")) {
-            this.getCounts();
-        }
-
-        if (diffState.hasAny("userSelectedTable", "counts")) {
-            const currentTable = this.state.userSelectedTable || this.state.counts && this.state.counts.resultsPerTable[0].tableName;
-            this.setState({ currentTable: currentTable });
-        }
-
-        if (diffState.hasAny("currentTable") && prevState.currentTable) {
+        if (diffProps.hasAny("currentTable") && prevProps.currentTable) {
             this.setState(this.getClearedUserSelections());

```

Comment:
Is this only when there was a previous table because you want to load the default query settings otherwise? I'm confused about why we wouldn't always clear when the currentTable changes.

Reply:
Short answer is it could probably go either way. Longer answer is that it should always get cleared on the way into the currentTable=undef state. And on startup the state is handled elsewhere.